data/common/__init__.py

```diff
@@ -212,3 +212,38 @@ def fix_unit(unit):
             return "t⋅km"
         case _:
             return unit
+
+
+def format_number(num):
+    """Format a number to a string using python general format and simplified exponential notation.
+
+    Args:
+        num: A number (int or float) to format
+
+    Returns:
+        str: Formatted number string with:
+        - 6 significant digits (using Python's general format)
+        - Simplified exponential notation to conform to prettier requirements where:
+            - "e+0" becomes "e" (e.g., "1.23e+05" → "1.23e5")
+            - "e-0" becomes "e-" (e.g., "1.23e-05" → "1.23e-5")
+            - "e+" becomes "e" (e.g., "1.23e+5" → "1.23e5")
+    """
+    if isinstance(num, (int, float)):
+        # Convert to scientific notation with 6 significant digits
+        num_py_g = f"{num:.6g}"
+        # Clean up the exponential notation
+        return float(
+            num_py_g.replace("e+0", "e").replace("e-0", "e-").replace("e+", "e")
+        )
+    return str(num)
+
+
+def format_numbers_recursively(data):
```

Comment:
The formatting can be done during the json encoding, using the cls argument of json.dump with something like that:
```
class CustomEncoder(json.JSONEncoder):
    def encode(self, obj):
        def format_numbers(obj):
            if isinstance(obj, (int, float)):
                return f"{obj:.6g}"
            elif isinstance(obj, dict):
                return {k: format_numbers(v) for k, v in obj.items()}
            elif isinstance(obj, list):
                return [format_numbers(v) for v in obj]
            else:
                return obj
        
        return super().encode(format_numbers(obj))
```
then
```
>>> json.dumps({"a": 1.23456789e-5, "b": [1.23456789e-5]}, cls=CustomEncoder)
'{"a": "1.23457e-05", "b": ["1.23457e-05"]}'
```

Reply:
I tried json.dump without success but indeed json.dumps() works indeed. Implemented, thanks